ebs/ebs.go

```diff
@@ -295,6 +295,20 @@ func (d *Driver) CreateVolume(req Request) error {
 		if err := d.ebsService.AddTags(volumeID, newTags); err != nil {
 			log.Debugf("Failed to update tags for volume %v, but continue", volumeID)
 		}
+
+		// detach from all instances
+		for _, attachment := range ebsVolume.Attachments {
+			// check the VM state before detaching , ensure no running instance is using the volume
+			running, err := d.ebsService.IsInstanceRunning(attachment.InstanceId)
+			if err != nil {
+				return err
```

Comment:
If an instance the ebs volume was attached has been deleted already, then code won't get into this loop in the first place. There is a race condition while running through this loop, though, we could resolve that by distinguishing the error code (saying this instance does not exist any more) that I can look into.